cf/test/test_Data.py

```diff
@@ -4316,6 +4316,36 @@ def test_Data__init__datetime(self):
         self.assertTrue((q == d).array.all())
         self.assertTrue((d == q).array.all())
 
+    def test_Data__str__(self):
+        """Test `Data.__str__`"""
+        elements0 = ("first_element", "last_element", "second_element")
+        for array in ([1], [1, 2], [1, 2, 3]):
```

Comment:
Ideally we could extend this test method to also include an example array input for `Data` with size larger than three to test that in such cases the `second_element` isn't included in `_custom` in this way and that an `Ellipsis` replaces it in the `str` representation.

Obviously it isn't as simple as adding it to this list to iterate over, but it it might be simple enough to adjust this block with some conditionals to allow that to work with some tweaks, else write a separate block to cover it?

Reply:
Excellent idea: https://github.com/NCAS-CMS/cf-python/pull/494/commits/71a8bf705e2927396aa1a32e56f146936cc3d524